tests/unittests/CoreText/CTFontTests.mm

```diff
@@ -435,4 +435,105 @@ virtual void TearDown() {
+}
+
+TEST(CTFont, GetBoundingBoxes) {
+// Font names differ slightly between platforms
+#if TARGET_OS_WIN32
+    CFStringRef name = CFSTR("Arial Bold Italic");
+#else
+    CFStringRef name = CFSTR("Arial-BoldItalicMT");
+#endif
+
+    CTFontRef font = CTFontCreateWithName(name, 585.45f, nullptr);
+    CFAutorelease(font);
+
+    const size_t count = 3;
+    CGGlyph glyphs[count] = { 68, 40, 22 };
+    CGRect boxes[count];
+
+    // Default/horizontal orientation
+    CGRect expectedBoxes[count] = {
+        { { 26.2995, -7.14661 }, { 285.864, 317.881 } },
+        { { 24.0126, 0 }, { 398.209, 419.077 } },
+        { { 29.7299, -7.14661 }, { 298.156, 427.939 } },
+    };
+
+    CGRect expectedTotalBox = { { 24.0126, -7.14661 }, { 398.209, 427.939 } };
+
+    CGRect totalBox = CTFontGetBoundingRectsForGlyphs(font, kCTFontDefaultOrientation, glyphs, boxes, count);
+    EXPECT_NEAR(expectedTotalBox.origin.x, totalBox.origin.x, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.origin.y, totalBox.origin.y, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.width, totalBox.size.width, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.height, totalBox.size.height, c_errorMargin);
+
+    for (size_t i = 0; i < count; ++i) {
+        EXPECT_NEAR(expectedBoxes[i].origin.x, boxes[i].origin.x, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].origin.y, boxes[i].origin.y, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].size.width, boxes[i].size.width, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].size.height, boxes[i].size.height, c_errorMargin);
+    }
+
+    totalBox = CTFontGetBoundingRectsForGlyphs(font, kCTFontHorizontalOrientation, glyphs, boxes, count);
+    EXPECT_NEAR(expectedTotalBox.origin.x, totalBox.origin.x, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.origin.y, totalBox.origin.y, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.width, totalBox.size.width, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.height, totalBox.size.height, c_errorMargin);
+    for (size_t i = 0; i < count; ++i) {
+        EXPECT_NEAR(expectedBoxes[i].origin.x, boxes[i].origin.x, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].origin.y, boxes[i].origin.y, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].size.width, boxes[i].size.width, c_errorMargin);
+        EXPECT_NEAR(expectedBoxes[i].size.height, boxes[i].size.height, c_errorMargin);
+    }
+
+    // Null outpointer for boxes should evaluate just the total box
+    totalBox = CTFontGetBoundingRectsForGlyphs(font, kCTFontHorizontalOrientation, glyphs, nullptr, count);
+    EXPECT_NEAR(expectedTotalBox.origin.x, totalBox.origin.x, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.origin.y, totalBox.origin.y, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.width, totalBox.size.width, c_errorMargin);
+    EXPECT_NEAR(expectedTotalBox.size.height, totalBox.size.height, c_errorMargin);
+
+// DWrite seems to not respect orientation
```

Comment:
Of course, lets not block this change while we sort that through.

---
In reply to: [83329813](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r83329813) [](ancestors = 83329813,83328388)